shuttle/src/runtime/execution.rs

```diff
@@ -663,20 +668,28 @@ impl ExecutionState {
         }
 
         let mut unfinished_attached = false;
-        let mut runnable = self
-            .tasks
-            .iter()
-            .inspect(|t| unfinished_attached = unfinished_attached || (!t.finished() && !t.detached))
-            .filter(|t| t.runnable())
-            .map(|t| t.id)
-            .collect::<SmallVec<[_; DEFAULT_INLINE_TASKS]>>();
+        let mut all_runnable_detached = true;
+        let mut has_runnable = false;
```

Comment:
nit: this should be `some_runnable` or `any_runnable` (not sure what the `has` refers to)